gateway/src/test/java/io/camunda/zeebe/gateway/RequestMapperTest.java

```diff
@@ -7,24 +7,102 @@
  */
 package io.camunda.zeebe.gateway;
 
+import static org.assertj.core.api.Assertions.assertThat;
 import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.assertj.core.api.Assertions.entry;
 
 import com.fasterxml.jackson.core.JsonParseException;
+import io.camunda.zeebe.gateway.protocol.GatewayOuterClass.ModifyProcessInstanceRequest;
+import io.camunda.zeebe.gateway.protocol.GatewayOuterClass.ModifyProcessInstanceRequest.ActivateInstruction;
+import io.camunda.zeebe.gateway.protocol.GatewayOuterClass.ModifyProcessInstanceRequest.TerminateInstruction;
+import io.camunda.zeebe.gateway.protocol.GatewayOuterClass.ModifyProcessInstanceRequest.VariableInstruction;
+import io.camunda.zeebe.protocol.impl.record.value.processinstance.ProcessInstanceModificationRecord;
 import org.junit.jupiter.api.Test;
 
 public class RequestMapperTest {
 
+  private static final String VALID_VARIABLES = "{ \"test\": \"value\"}";
+  // missing closing quote in second variable
+  private static final String INVALID_VARIABLES =
+      "{ \"test\": \"value\", \"error\": \"errorrvalue }";
+
   @Test
   public void shouldThrowHelpfulExceptionIfJsonIsInvalid() {
+    // when + then
+    assertThatThrownBy(() -> RequestMapper.ensureJsonSet(INVALID_VARIABLES))
+        .isInstanceOf(JsonParseException.class)
+        .hasMessageContaining("Invalid JSON", INVALID_VARIABLES)
+        .cause()
+        .isInstanceOf(JsonParseException.class);
+  }
+
+  @Test
+  public void shouldMapRequestToModifyProcessInstanceRequest() {
```

Comment:
:wrench: We could extract the test cases for the modification request into its own class. 

For the other commands, there are separated test classes. For example [SetVariablesTest.java](https://github.com/camunda/zeebe/blob/main/gateway/src/test/java/io/camunda/zeebe/gateway/api/process/SetVariablesTest.java).

Otherwise, the `RequestMapperTest` could grow too much in the future if we add new requests.